base/src/schema.test.ts

```diff
@@ -915,6 +927,213 @@ describe('schema', () => {
 
       expect(() => Schema.validateSchema(schema)).not.toThrow()
     })
+
+    it('throws an error when schema definitions is not an object', () => {
+      const schema = [{ stuffs: { type: 'array' } }]
+
+      expect(() =>
+        Schema.validateSchema(schema as unknown as SchemaDefinitions)
+      ).toThrow('schema has to be an object')
+    })
+
+    it('throws a validation error when type is unknown', () => {
+      const schema = {
+        stuffs: { type: '' },
+      }
+
+      expect(() =>
+        Schema.validateSchema(schema as unknown as SchemaDefinitions)
+      ).toThrow('`type` is invalid or undefined')
+    })
+
+    it('throws a validation error when items isnt defined for an array', () => {
+      const schema = {
+        stuffs: { type: 'array' },
+      }
+
+      expect(() =>
+        Schema.validateSchema(schema as unknown as SchemaDefinitions)
+      ).toThrow('`items` is required when type is `array`')
+    })
+
+    it('throws a validation error when defaultValue is not an array', () => {
+      const schema = {
+        stuff: {
+          defaultValue: 'name',
+          items: { defaultValue: '5', type: 'string' },
+          type: 'array',
+        },
+      }
+
+      expect(() =>
+        Schema.validateSchema(schema as unknown as SchemaDefinitions)
+      ).toThrow('`defaultValue` should be an array')
+    })
+
+    it('validates an array schema', () => {
```

Comment:
? 

It says "validates an array schema" but checks a `.toThrow` without any message.

It's unclear what this test is doing